core/jobs/io/gcs_io_test.py

```diff
@@ -0,0 +1,175 @@
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(filepaths)
+            | 'Write to GCS' >> gcs_io.WriteFile()
+        )
+        self.assert_pcoll_equal(filepath_p_collec, [3681])
+
+
+class DeleteFileTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.DeleteFile."""
+
+    def test_delete_files_in_gcs(self) -> None:
+        file_path = 'dummy_folder/dummy_subfolder/dummy_file'
+        string = b'testing'
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        storage_services.commit(bucket, file_path, string, None)
+        file_paths = [file_path]
+        filepath_p_collec = (
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(file_paths)
+            | 'Delete file from GCS' >> gcs_io.DeleteFile()
+        )
+        self.assert_pcoll_equal(filepath_p_collec, [None])
+
+    def test_check_correct_files_are_passing(self) -> None:
+        file_path = 'dummy_folder/dummy_subfolder/dummy_file'
+        file_paths = [file_path]
+
+        # Here we use MyPy ignore because it raises an error for not having
+        # typing for "_". The underscore is representing the "self" which
+        # is getting passed by the gcs_io.
+        def _mock_delete(_, filepath: str) -> str: # type: ignore[no-untyped-def]
+            return filepath
+
+        with self.swap(gcs_io.DeleteFile, '_delete_file', _mock_delete):
+            filepath_p_collec = (
+                self.pipeline
+                | 'Create pcoll of filepaths' >> beam.Create(file_paths)
+                | 'Delete file from GCS' >> gcs_io.DeleteFile()
+            )
+            self.assert_pcoll_equal(filepath_p_collec, [file_path])
+
+
+class GetFilesTest(job_test_utils.PipelinedTestBase):
+    """Tests to check gcs_io.GetFiles."""
+
+    def test_get_files_with_specefic_prefix(self) -> None:
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        filepath_1 = 'dummy_folder/dummy_subfolder/dummy_file_1'
+        filepath_2 = 'dummy_folder/dummy_subfolder/dummy_file_2'
+        string = b'testing'
+        storage_services.commit(
+            bucket, filepath_1, string, 'application/octet-stream')
+        storage_services.commit(
+            bucket, filepath_2, string, 'application/octet-stream')
+        emulator_blob_1 = cloud_storage_emulator.EmulatorBlob(
+            filepath_1, string, 'application/octet-stream')
+        emulator_blob_2 = cloud_storage_emulator.EmulatorBlob(
+            filepath_2, string, 'application/octet-stream')
+        prefixes = ['dummy_folder/dummy_subfolder']
+        filepath_p_collec = (
+            self.pipeline
+            | 'Create pcoll of filepaths' >> beam.Create(prefixes)
+            | 'Get files from GCS' >> gcs_io.GetFiles()
+        )
+        self.assert_pcoll_equal(
+            filepath_p_collec, [
+                [emulator_blob_1, emulator_blob_2]
+            ])
+
```

Comment:
Hey @vojtechjelinek,
I'm done with the new implementation of GCS Io and things are working perfectly fine. Just one nit, this test will fail and the reason is the sequence, when we do `listdirs` from storage services it gives the result in any sequence so sometimes the tests pass and sometimes they don't. Please let me know what I can I do here.
Everything else is done and tested.

Reply:
Order shouldn't matter for `assert_pcoll_equal` I think?

Reply:
It was getting returned in the list and the order was not matching inside the returned list, I have updated this part and now the tests are passing.